### DISCUSSION TITLE
Can tabs be disabled?

### DISCUSSION THREAD

**Discussion by mightyiam** [Jelmerro/Vieb#400] [Q&A] [answered]
I'm looking for a tab-less browsing experience. Can tabs be disabled completely, please?

**Comment by Jelmerro** [comment, marked as answer]
Depending on how hard you want to disable them, you can. Allow me to explain. There is no hard switch or single setting to disable tabs completely. However, you could start with hiding the tabbar, using `:set guitabbar=never`. This will not change how tabs work, but hide them from the UI completely. Secondly, you can change every single mapping in Vieb with it's mapping system, see `:h mappings` for details, but as an example, you can remove the newtab mapping with `:nunmap t`. You could also map these keys to different actions now that they are available, but that's up to you. Last but not least, you can control which container new tabs should open with `containernewtab`, which also has the option to open new tabs externally or in the current tab always, either with `:set containernewtab=s:external` or `:set containernewtab=s:replacecurrent` respectively. Hiding the tab bar and not allowing new tabs to be opened sounds like "completely disabling tabs", but it really just depends on how strictly you want them disabled. Hopefully this answered your question, if not, please do not hesitate to ask more followup questions in the comments below.

**Comment by v3ss0n** [comment]
That is so awesome thank you so much.
One small thing : I want to open new window inplace of tabs , what should i do ? s:external leads to other browser , not as new vieb window.

**Reply by Jelmerro** [reply]
Glad you like it. Electron and by extension Vieb do not support multiple windows properly, so Vieb is a single-window application, hence the external option. You can hack it back in the code, but expect your cache, cookies etc to not be shared or even corrupted if you do. Sorry, it's just what it is.

**Reply by v3ss0n** [reply]
i see , i just relaised that limitation ,yeah that make sense , i tried doing external and setting vieb as default browser but it doesn't work , it just replaces current. . `s:replacecurrent` have one problem : for pinned tabs , they will replace pinned tabs which is not good (i pinned my major communication/work related stuffs there , and replace current override them , any way to do so opening new tabs works on pinned tab but not on others? , Or how about setting for opening new links as `:split` `:vsplit` instead of tabs?

**Reply by Jelmerro** [reply]
There is no container setting for it, but you can open links in new splits or vertical splits with these follow mode variants: `startFollowNewSplit` and `startFollowNewVerSplit`.
